lib/gcapp.class.php

```diff
@@ -392,14 +396,12 @@ public static function refreshMapfile($project, $mapset, $publish = false, $refr
 		$alternativeLanguages = $localization->getAlternativeLanguages();
 		if($alternativeLanguages){
 			foreach($alternativeLanguages as $languageId => $foo) {
-				$mapfile = new gcMapfile($languageId);
-				$mapfile->setTarget($target);
+				$mapfile = new gcMapfile($languageId, $target);
 				$mapfile->writeMap('mapset', $mapset);
                 
                 if ($refreshLayerMapfile) {
                     foreach(GCAuthor::getLayerList($project, $mapset) as $mapsetData) {
-                        $mapfile = new gcMapfile();
-                        $mapfile->setTarget('layer');
+                        $mapfile = new gcMapfile($languageId, "layer");

```

Comment:
Yes, this is a fix. Before there were no translated mapfiles for single layers, it would overwrite the same mapfiles it just created in the loop above without translation (So twice execution time for every layer/language combination with 0 benefit). Now it correctly creates a translated mapfile for every translation and layer.